### DISCUSSION TITLE
Adding more tag sources, plus a discrete overwrite option.

### DISCUSSION THREAD

**Discussion by rosgr100** [Marekkon5/onetagger#228] [Ideas]
Hi @Marekkon5, thanks for the amazing work you have done till now.

i was wondering if it's possible to add more source to the AT, such as BPMlatino (this must be the easiest due to the existence of BPMsupreme), Djcity which gives free access to their tags, or Mymp3pool (needs a free account). 

Also, is it possible to have a discrete Overwrite tag option per tag or at least for the important ones like, title, artist, bpm and key?

Thank you.

**Comment by Marekkon5** [comment]
Added BPMLatino, will look at rest when I have more time. Also you can make your own platforms using: https://github.com/Marekkon5/onetagger-platform-template

**Comment by rosgr100** [comment]
Thanks for the effort @Marekkon5.

Yes, you are right, at some point i need to take a look on the Custom Platform to find out how to do it.

**Comment by Marekkon5** [comment]
Added overwrite per tag in latest commit. You can get the build from Actions tab

**Comment by rosgr100** [comment]
Thank you for the effort you have put to satisfy one more of requests @Marekkon5

**Comment by MatoLabu** [comment]
I'm not sure to understand...
Is there a way to prevent One Tagger from overwriting certain tags (BPM for me) when saving?

Explanation:

I use Traktor 4 with stems (.m4a files) and I’d love to integrate One Tagger into my workflow. Until now, I’ve been using Foobar2000 and managing my playlists as .m3u files, but I’d like to switch to One Tagger and take advantage of smartlists.  

Here’s my process:  
1. I tag my files in One Tagger – everything works fine.  
2. I add the files to my Traktor collection and analyze them.  
3. Traktor detects a BPM with decimals (e.g., 119.14 BPM).  
4. Later, I want to edit a tag (e.g., the genre) in One Tagger.  
5. After saving in One Tagger and reopening Traktor, the BPM is now rounded (119.14 → 119.00).  

I assume One Tagger does not store BPM values with decimals, which isn’t a big issue since I rely on Traktor’s BPM detection and manually refine decimals if needed.  

However, my concern is that saving any tag modification in One Tagger overwrites the BPM, which means I can’t edit my tags without losing BPM precision in Traktor. Is there a way to prevent One Tagger from modifying the BPM field when saving?  

I see this topic has been discussed above, but I didn’t fully understand the solution—if there is one. Without a fix, I wouldn’t be able to use One Tagger in my workflow, which would be a real shame!  

Thanks in advance, and I hope my explanation is clear.  

Best,  
MatoLabu

**Reply by MatoLabu** [reply]
In case it's useful, I tested several tagging software and found that TagScanner can modify a tag (e.g., adding 'Funk' to the genres) without altering the BPM, preserving Traktor's decimal values. In contrast, Mp3Tag appears to save all fields, rounding the BPM to an integer, similar to One Tagger 1.7.0.

**Reply by MatoLabu** [reply]
After extensive testing, here’s my analysis of Traktor’s behavior:

- Suppose I set the BPM to 123.00 in Mp3Tag.
- Then, in Traktor, I manually adjust the BPM to 123.45.
- Comparing the hex values (using 010 Editor), there is no difference, meaning the decimal values do not appear to be stored in the track file itself.

From there, I tested various scenarios using TagScanner (which seems to allow modifying a tag without altering anything else):

- Scenario A: If I change the BPM from 123.45 to 124.00, Traktor correctly registers 124.00.
- Scenario B: If I change the BPM from 123.45 to 0, Traktor ignores the change and keeps 123.45.
- Scenario C: If I change the BPM from 123.45 to 199, save the file, then change it back to 123.00 (meaning the file is now exactly the same as before editing with TagScanner), Traktor detects no change, so the BPM remains 123.45.

My question is:

Would it be possible for OneTagger’s QuickTag feature to behave like TagScanner (modifying only the tags that actually change), instead of acting like Mp3Tag, which appears to update all tags and, in doing so, seems to cause Traktor to lose its decimal BPM values?

I assume this issue affects all Traktor users, so implementing this behavior would be a major improvement for many people. 🙏

**Reply by MatoLabu** [reply]
### Solution Found!

After days of research, testing multiple tag editing softwares, analyzing Traktor’s behavior, and even cloning OneTagger's source code in an attempt to understand and modify it myself (I’m not a coder)… I finally found a solution!

And it turns out to be incredibly simple, yet I had never used this function before:

- In Traktor, after analyzing the BPM with decimals (or manually fine-tuning them), simply **right-click the track** > **"Enable Analysis Lock"**.
- This option is also available when fully expanding a **track deck**, switching to the **"grid" view**, and clicking the **small lock icon** to the left of the BPM.
- With this enabled, you can use **any** tag editor (including OneTagger) to modify tags, and when you load the track back into Traktor, **the BPM remains exactly as set—down to the hundredth decimal!**

I’m beyond relieved!

Bonus fix:
I also had an issue where **M4A/MP4 tracks containing stems wouldn’t play properly in OneTagger’s built-in player**. I solved this by setting OneTagger to open tracks in an external player, and now my files play seamlessly in **Winamp**, with its convenient **keyboard shortcuts** for seeking and playback—**without losing focus** on OneTagger window.

Problem solved, dream workflow achieved! 🎉
